### ISSUE TITLE
Ability to generate self-organizing Metric Tree 

### ISSUE THREAD

**Issue by ockhamlabs** [DataJunction/dj#1137]
Have you considered / already use the semantic layer to build metric trees? https://www.leverslabs.com/article/introducing-metric-trees  (e.g. of what it is)
Super helpful in understanding relationships and evaluating impacts. I suspect given the approach you are using there might be a self-organizing metric capability that could exist with platform. Alternatively, it could be a field manually identified where a user can flag if a particular metric has a parent and if so which (one or more may be too)

**Comment by shangyian** [comment]
Hi @ockhamlabs, we already generate graphs for each node in the graph view tab in DJ. The API also provides more information on which parent a metric has. Are you looking for something additional that's in the concept of a metric tree?

**Comment by vcknorket** [comment]
I see...so we can construct a metric tree by identifying the parents ?

**Comment by shangyian** [comment]
@vcknorket yep, you should be able to do that just by calling `GET /nodes/growth.xp.funnel.nonmember_funnel_base/upstream`, which will return all upstream nodes.

**Comment by vcknorket** [comment]
[like]  Venkata Chandra (VC) reacted to your message:
________________________________

**Comment by vcknorket** [comment]
Thanks

________________________________
